File: python_programs/thread_creation.py
import time
import threading

NUM_THREADS = 1000
total_time = 0

def dummy_function():
    pass

for _ in range(10):
    start = time.perf_counter()
    threads = [threading.Thread(target=dummy_function) for _ in range(NUM_THREADS)]
    for t in threads:
        t.start()
    for t in threads:
        t.join()
    end = time.perf_counter()
    total_time += (end - start) * 1e9

with open("./results/python_thread_creation_results.txt", "w") as f:
    f.write(f"Average: {total_time / 10:.6f} ns\n")

File: java_programs/ThreadCreation.java
import java.io.FileWriter;
import java.io.IOException;

public class ThreadCreation {
    public static void main(String[] args) throws IOException {
        int NUM_THREADS = 1000;
        long totalTime = 0;

        for (int test = 0; test < 10; test++) {
            long start = System.nanoTime();
            Thread[] threads = new Thread[NUM_THREADS];
            for (int i = 0; i < NUM_THREADS; i++) {
                threads[i] = new Thread(() -> {});
                threads[i].start();
            }
            for (int i = 0; i < NUM_THREADS; i++) {
                try {
                    threads[i].join();
                } catch (InterruptedException e) {
                    e.printStackTrace();
                }
            }
            long end = System.nanoTime();
            totalTime += (end - start);
        }

        FileWriter writer = new FileWriter("./results/java_thread_creation_results.txt");
        writer.write(String.format("Average: %.6f ns\n", totalTime / 10.0));
        writer.close();
    }
}
